spec/models/membership_application_spec.rb

```diff
@@ -140,22 +149,27 @@
 
   end
 
-  describe "#destroy" do
+  describe 'destroy callbacks' do
 
-    let(:application_owner) {create(:user, email: 'user_1@random.com')}
-    let(:uploaded_file) {create(:uploaded_file, actual_file: (File.new(File.join(FIXTURE_DIR, 'image.jpg'))))}
-    let(:membership_application) {create(:membership_application, user: application_owner, uploaded_files: [uploaded_file])}
+    let(:application_owner) { create(:user, email: 'user_1@random.com') }
+
+    app_file = (File.new(File.join(FIXTURE_DIR, 'image.jpg')))
+    let(:uploaded_file) { create(:uploaded_file, actual_file: app_file) }
 
-    it 'destroys a membershipapplication' do
-      membership_application.destroy
-      expect(membership_application.destroyed?).to be_truthy
+    let(:application) do
+      create(:membership_application, user: application_owner,
+             uploaded_files: [uploaded_file], state: :accepted)
     end
 
-    it 'destroys the uploaded file' do
-      membership_application.destroy
+    it 'invokes method to delete uploaded files' do
+      application.destroy
       expect(uploaded_file.destroyed?).to be_truthy
     end
 
+    it "destroys associated company" do
```

Comment:
I'd expand the description to "destroys the associated company if it's the only application"